src/psyclone/psyGen.py

```diff
@@ -969,6 +970,27 @@ def __init__(self, ast=None, children=None, parent=None):
     def __str__(self):
         raise NotImplementedError("Please implement me")
 
+    def math_equal(self, other):
+        '''Returns true if the self has the same results as other. The
+        implementation in the base class just confirms that the type is the
+        same, and the number of children as well.
+
+        :param other: the node to compare self with.
+        :type other: py:class:`psyclone.psyGen.Node`.
+        '''
+
+        # pylint: disable=unidiomatic-typecheck
+        if type(self) != type(other):
+            return False
+
+        if len(self.children) != len(other.children):
+            return False
+
+        for i, entity in enumerate(self.children):
+            if not entity.math_equal(other.children[i]):
```

Comment:
I did notice your comment, but I still need to clarify this: this code is very limited in its scope. E.g. the expressions 'i+n+1' and 'i+1+n' would likely not be considered equal (assuming that '+' binds left first: (i+n)+1 and (i+1)+n. We could of course extend it, but the question is how much effort do we want to spend in this? At a certain stage it might be simpler to plug in an (optional??) symbolic maths package - afaik something like this exists for python.

Should we perhaps even remove this option for now, considering that it is limited? Or extend it to work at least for 'n' identical binary operations (e.g. i+n+1 as above)? The moment we support more than one operation (i+5*k+1 vs i+1+k*5 ) ... it becomes a lot more complicated.

Reply:
OK, I've created #533, documented the current behaviour in the developer guide, added more test cases that show what works, and an xfailing one with a reference to #533 showing what does not work.

Reply:
You're a star, thanks.